sbi/samplers/rejection/rejection.py

```diff
@@ -257,35 +257,28 @@ def accept_reject_sample(
     num_sampled_total, num_remaining = 0, num_samples
     accepted, acceptance_rate = [], float("Nan")
     leakage_warning_raised = False
+    # Ruff suggestion
+    if proposal_sampling_kwargs is None:
+        proposal_sampling_kwargs = {}
 
     # To cover cases with few samples without leakage:
     sampling_batch_size = min(num_samples, max_sampling_batch_size)
     while num_remaining > 0:
         # Sample and reject.
-        # This if-case is annoying, but it will be gone when we move away from
-        # nflows and towards a flows-framework which takes a tuple as sample_size.
-        if isinstance(proposal, nn.Module):
-            candidates = proposal.sample(
-                sampling_batch_size,
-                **proposal_sampling_kwargs or {},  # type: ignore
-            ).reshape(sampling_batch_size, -1)
-        else:
-            candidates = proposal.sample(
-                torch.Size((sampling_batch_size,)),
-                **proposal_sampling_kwargs or {},  # type: ignore
-            )  # type: ignore
+        candidates = proposal.sample(
+            (sampling_batch_size,),  # type: ignore
+            **proposal_sampling_kwargs,
+        )
 
         # SNPE-style rejection-sampling when the proposal is the neural net.
         are_accepted = accept_reject_fn(candidates)
-
         samples = candidates[are_accepted]
-
         accepted.append(samples)
 
         # Update.
         num_sampled_total += sampling_batch_size
-        num_remaining -= samples.shape[0]
-        pbar.update(samples.shape[0])
+        num_remaining -= samples.shape[-2]
```

Comment:
Thanks for the explanation Jan, I would have had the same question. Let's please add an inline comment saying this.